src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
+          
+    // determines the maximal level at which a target occurs
+    let rec determineLevels (currentLevel : int) (levels : Map<string, int>) (t : seq<TargetTemplate<unit>>) =
+        let levels = 
+            t |> Seq.fold (fun m t -> 
+                match Map.tryFind t.Name m with
+                    | Some mapLevel when mapLevel >= currentLevel -> m
+                    | _ -> Map.add t.Name currentLevel m
+            ) levels
+            
+        t |> Seq.map (fun t -> t.Dependencies |> Seq.map getTarget) 
+          |> Seq.fold (determineLevels (currentLevel + 1)) levels
+
+    let levels = determineLevels 0 Map.empty [d]
+
+    // the results are grouped by their level, sorted descending (by level) and 
+    // finally grouped together in a list<TargetTemplate<unit>[]>
+    let result = 
+        levels |> Map.toSeq
+               |> Seq.groupBy snd
+               |> Seq.sortBy (fun (l,_) -> -l)
+               |> Seq.map snd
+               |> Seq.map (fun v -> v |> Seq.map fst |> Seq.distinct |> Seq.map getTarget |> Seq.toArray)
+               |> Seq.toList
+
+    // Note that this build order cannot be considered "optimal" 
+    // since it may introduce order where actually no dependencies
+    // exist. However it yields a "good" execution order in practice.
+    result
+
+/// Runs a single target without its dependencies
+let runSingleTarget (target : TargetTemplate<unit>) =
+    try
+        if errors = [] then
+            traceStartTarget target.Name target.Description (dependencyString target)
+            let watch = new System.Diagnostics.Stopwatch()
+            watch.Start()
+            target.Function()
+            addExecutedTarget target.Name watch.Elapsed
+            traceEndTarget target.Name     
+    with exn ->
+        targetError target.Name exn
+
+/// Runs the given array of targets in parallel using count tasks
+let runTargetsParallel (count : int) (targets : Target[]) =
```

Comment:
It turns out that the both LINQ solutions uses the thread from which they are executed as one of the worker threads. This is what it's like if it's run from main thread (on a 4 cores box):

![image](https://cloud.githubusercontent.com/assets/873919/6547452/9e5068cc-c5e9-11e4-9d9a-abf555b03373.png)

I think it's totally ok for this case, but you should understand that not all targets will be executed in Thread Pool.